tortoise/algorithm.go

```diff
@@ -147,6 +146,20 @@ func (t *Tortoise) LatestComplete() types.LayerID {
 	return t.trtl.verified
 }
 
+func (t *Tortoise) Updates() (types.LayerID, []types.BlockContextualValidity) {
```

Comment:
why it needs to return verified layer? can mesh work just based on updates?

Reply:
it definitely correct to work on the list of updates, originally i proposed this type to track empty layers. but map[Layer][]Block also works

```go
type Event struct {
    Layer uint32
    []BlockHeader
}
```

Reply:
what i want to achieve is to get a consistent view of tortoise, as now there are two separate threads that can cause verified layer to change (proposal builder and mesh).

so here verified layer and the list of updates are atomic, the information also feeds to `getLayerValidBlocks` to pick the block to apply.

Reply:
my point was that there is no need to read verified layer at all. the least of updates should be enough for mesh

Reply:
yes. i can usually work out the verified layer from the list of updates. but for empty layer that is tricky.
say
t0: at layer 112 hare failed (network flakiness or whatever)
t1: at layer 113 tortoise verified layer 112 to be an empty layer, but the list of updates would be empty

the mesh won't be able to determine that it needs to apply an empty layer without knowing that tortoise actually has verified to 112.
(until later synced a certificate for the EmptyBlock ID, but this is out-of-protocol)

Reply:
i don't understand what is the problem
there will be a list: [{111, [{a, true}],} {112, []}]
111 is not empty, block a is valid
112 is empty, no known blocks

also i am not sure how current variant works
verified layer can move back if expected weight increased, but actual weight didn't

Reply:
oh. i had thought that {112, []} will _not_ be in the updates from one of our previous discussion.
ok. in that case i will drop the verified layer from the API. thanks.

Reply:
i just realized, while changing the code, that i need to know the latest verified layer anyway if tortoise didn't make any new progress (i.e. empty updates)
previously this is accomplished by calling `tortoise.LatestComplete()`. here i condense it in one call `tortoise.Updates()`.
do you object to leaving the code as is? @dshulyak

Reply:
anyhow, i'll merge as is. please file an issue if you disagree.

Reply:
sorry i was away when you posted last messages

> here i condense it in one call tortoise.Updates().
do you object to leaving the code as is?

i still think that this is not needed. tortoise can be tweaked to emit event when layer is terminated as empty. i also wrote it above